keel-core/src/main/kotlin/com/netflix/spinnaker/keel/core/api/SubmittedResource.kt

```diff
@@ -36,10 +36,10 @@ data class SubmittedResource<T : ResourceSpec>(
   @Description("The specification of the resource")
   @JsonTypeInfo(use = Id.NAME, include = As.EXTERNAL_PROPERTY, property = "kind")
   val spec: T
-)
-
-val <T : ResourceSpec> SubmittedResource<T>.id: String
-  get() = generateId(kind, spec)
+) {
+  val id: String
```

Comment:
This avoids the need to import the `id` extension all over the place, especially since it was already defined within the class itself and didn't have any external dependencies.

Reply:
Are you sure this isn't going to require users to specify the id on submission?

Also you could use `val id: String by lazy { generateId(kind,  spec) }` since you don't really need to re-generate the id every time someone calls `mySubmittedResource.id`

Reply:
Ah. Good question. I'll just revert that. Not worth the risk.